src/mavlink-router/mainloop.cpp

```diff
@@ -564,12 +634,28 @@ Timeout *Mainloop::add_timeout(uint32_t timeout_msec, std::function<bool(void*)>
 
 error:
     delete t;
-    return NULL;
+    return nullptr;
+}
+
+void Mainloop::set_timeout(Timeout *t, uint32_t timeout_msec)
+{
+    if (!t) {
+        return;
+    }
+
+    struct itimerspec ts;
+    ts.it_interval.tv_sec = timeout_msec / MSEC_PER_SEC;
+    ts.it_interval.tv_nsec = (timeout_msec % MSEC_PER_SEC) * NSEC_PER_MSEC;
+    ts.it_value.tv_sec = ts.it_interval.tv_sec;
+    ts.it_value.tv_nsec = ts.it_interval.tv_nsec;
+    timerfd_settime(t->fd, 0, &ts, nullptr);
```

Comment:
This should be a method of Timeout, not the mainloop. add_timeout() is the only method that is capable of creating a Timeout object since it also adds it to the mainloop, but setting a timeout should be perfectly possible from the Timeout object itself.